e3sm_to_cmip/util.py

```diff
@@ -306,195 +359,389 @@ def print_var_info(handlers, freq=None, inpath=None, tables=None, outpath=None):
+        default_handlers[key]["raw_variables"] = [
+            default_handlers[key].pop("e3sm_name")
+        ]
+        # Rename "cmip6_name" key to "name"
+        default_handlers[key]["name"] = default_handlers[key].pop("cmip_name")
+
+    return default_handlers
+
+
+def _get_complex_handlers(handlers_path: str) -> Dict[str, Dict[str, str]]:
+    """Gets the information for all complex handlers in a nested dictionary.
+
+    The parent key is the CMIP name of the variable, and the value is a nested
+    dictionary storing key/value pairs for "name", "units", "table",
+    "method", "raw_variables", "positive" (optional), and "levels" (optional).
+
+    Example output:
+    ---------------
+    {
+        "cesm_mmrbc": {
+            "name": "mmrbc",
+            "units": "kg kg-1",
+            "table": "CMIP6_AERmon.json",
+            "method": "<function cesm_mmrbc.handle(infiles, tables, user_input_path, **kwargs)>",
+            "raw_variables": ["bc_a1", "bc_a4", "bc_c1", "bc_c4"],
+            "positive": None,
+            "levels": {
+                "name": "standard_hybrid_sigma",
+                "units": "1",
+                "e3sm_axis_name": "lev",
+                "e3sm_axis_bnds": "ilev",
+            },
+        }
+    }
+
+    Parameters
+    ----------
+    handlers_path : str
+        The path to the complex handlers, which are Python modules.
+
+    Returns
     -------
-        handlers_path (str): the path to the python module to load handlers from
-        var_list (list(str)): A list of strings with the names of cmip6 variables to convert to, optionally "all" to run all handlers found
-    Returns:
-    --------
-        handlers (list(dict()): A list of dictionaries mapping module names
-        (which are the cmip6 output variable name), to a tuple of (function pointer,
-        list of required input variables)
+    Dict[str, Dict[str, str]]
+        A dictionary of dictionaries, with each dictionary storing the
+        information for a complex handler.
     """
-    from e3sm_to_cmip.default import default_handler
+    complex_handlers = {}
+
+    for root, _, files in os.walk(handlers_path):
+        for file in files:
+            if file.endswith(".py") and file != "__init__.py":
+                var = file.split(".")[0]
+                filepath = os.path.join(root, file)
+                module = imp.load_source(var, filepath)
+
+                complex_handlers[var] = {
+                    "name": module.VAR_NAME,
+                    "units": module.VAR_UNITS,
+                    "table": module.TABLE,
+                    "method": module.handle,
+                    "raw_variables": module.RAW_VARIABLES,
+                    "positive": module.POSITIVE
+                    if hasattr(module, "POSITIVE")
+                    else None,
+                    "levels": module.LEVELS if hasattr(module, "LEVELS") else None,
+                }
 
-    handlers = list()
+    return complex_handlers
```

Comment:
This function checks the `/cmor_handlers `(var `handlers_path`) directory for Python modules related to the "complex" handlers. It loads the Python modules to extract information from each module related to a handler.